providers/elasticsearch/src/airflow/providers/elasticsearch/hooks/elasticsearch.py

```diff
@@ -268,3 +282,448 @@ def search(self, query: dict[Any, Any], index: str = "_all") -> dict:
         es_client = self.get_conn
         result = es_client.search(index=index, body=query)
         return result["hits"]
+
+
+class ElasticsearchHook(BaseHook):
+    """
+    Hook for interacting with Elasticsearch.
+
+    This hook provides a convenient interface to Elasticsearch operations
+    including search, bulk operations, and data management.
+    """
+
+    conn_name_attr = "elasticsearch_conn_id"
+    default_conn_name = "elasticsearch_default"
+    conn_type = "elasticsearch"
+    hook_name = "Elasticsearch"
+
+    def __init__(
+        self, elasticsearch_conn_id: str = "elasticsearch_default", log_query: bool = False, **kwargs: Any
```

Comment:
Optional: Instead of hardcoding `"elasticsearch_default"`, this could fall back to the class-level 
`default_conn_name`. That way the default only needs to be maintained in one place.

```suggestion
        self, elasticsearch_conn_id: str = None, log_query: bool = False, **kwargs: Any
```